source/NVDAObjects/behaviors.py

```diff
@@ -309,25 +352,65 @@ def _monitor(self):
 			self._event.clear()
 
 			try:
-				newLines = self._getTextLines()
+				newText = self._getText()
 				if config.conf["presentation"]["reportDynamicContentChanges"]:
-					outLines = self._calculateNewText(newLines, oldLines)
+					outLines = self._calculateNewText(newText, oldText)
 					if len(outLines) == 1 and len(outLines[0].strip()) == 1:
 						# This is only a single character,
 						# which probably means it is just a typed character,
 						# so ignore it.
 						del outLines[0]
 					if outLines:
 						queueHandler.queueFunction(queueHandler.eventQueue, self._reportNewLines, outLines)
-				oldLines = newLines
+				oldText = newText
 			except:
-				log.exception("Error getting lines or calculating new text")
+				log.exception("Error getting or calculating new text")
 
-	def _calculateNewText(self, newLines, oldLines):
+		if self.shouldUseDMP:
+			try:
+				self._terminateDMP()
+			except Exception:
+				log.exception("Error stopping DMP")
+
+	def _calculateNewText_dmp(self, newText: str, oldText: str) -> List[str]:
+		try:
+			if not newText and not oldText:
+				# Return an empty list here to avoid exiting
+				# nvda_dmp uses two zero-length texts as a sentinal value
+				return []
+			old = oldText.encode("utf-8")
+			new = newText.encode("utf-8")
+			tl = struct.pack("=II", len(old), len(new))
+			self._dmp.stdin.write(tl)
+			self._dmp.stdin.write(old)
+			self._dmp.stdin.write(new)
+			buf = b""
+			sizeb = b""
+			SIZELEN = 4
+			while len(sizeb) < SIZELEN:
+				sizeb = self._dmp.stdout.read(SIZELEN - len(sizeb))
+				if sizeb is None:
+					sizeb = b""
+			(size,) = struct.unpack("=I", sizeb)

```

Comment:
extracting from `behaviors.py` would allow code like this to be unit tested more easily. This code looks suspicious to me, I'd like to be able to consult the unit tests to confirm it's behavior. The question that comes to mind is "is there a race condition? This looks like read may return less bytes than it's args, is that true? Is `sizeb` overwritten with each read, so if this reads first a single byte, it will then subsequently overwrite with just 3 bytes of the integer."

Some other concerns that aren't explicitly covered with code or comments explaining why it's not necessary:
- What about byte order?
- What about sizeof(int) mismatch?

Neither of these are a concern because the other end of the pipe is written in python and running the same version of the interpreter.